### PR TITLE
HEEDLS-447 - Implement display and editing of ShowOnMap for centres

### PR DESCRIPTION
**Pull request by DanBloxham-sw**
- Ran all unit tests
- Tested manually in Chrome, IE11, Edge, Firefox
- Tested screen reader/zoom/mobile view/no JS

Screenshots if new UI additions:
/TrackingSystem/CentreConfiguration page (Centre content on DLS website expandable):

Centre Config index expandable

/TrackingSystem/CentreConfiguration/EditCentreWebsiteDetails page:
Centre Config edit details

### CODE REVIEW

File: DigitalLearningSolutions.Web/Views/TrackingSystem/CentreConfiguration/EditCentreWebsiteDetails.cshtml [hunk 1]
```diff
@@ -12,23 +11,13 @@
...
-                                                   nameof(Model.CentreEmail),
-                                                   nameof(Model.CentrePostcode),
-                                                   nameof(Model.OpeningHours),
-                                                   nameof(Model.CentreWebAddress),
-                                                   nameof(Model.OrganisationsCovered),
-                                                   nameof(Model.TrainingVenues),
-                                                   nameof(Model.OtherInformation)
-                                                 })" />
+    @if (errorHasOccurred) {
+      <vc:error-summary order-of-property-names="@(new[] { nameof(Model.CentreTelephone), nameof(Model.CentreEmail), nameof(Model.CentrePostcode), nameof(Model.ShowCentreOnMap), nameof(Model.OpeningHours), nameof(Model.CentreWebAddress), nameof(Model.OrganisationsCovered), nameof(Model.TrainingVenues), nameof(Model.OtherInformation) })"/>
```
Comment (AlexJacksonDS) [inline DigitalLearningSolutions.Web/Views/TrackingSystem/CentreConfiguration/EditCentreWebsiteDetails.cshtml:20]: Can you sort out the formatting here. Looks like the auto formatter doesn't like this setup.

**Review by AlexJacksonDS — APPROVED** [review]
Looks fine just one comment about formatting

File: DigitalLearningSolutions.Data.Tests/DataServices/CentresDataServiceTests.cs [hunk 2]
```diff
@@ -208,6 +212,7 @@ public void UpdateCentreWebsiteDetails_updates_centre()
                     updatedCentre.CentreTelephone.Should().BeEquivalentTo(telephone);
                     updatedCentre.CentreEmail.Should().BeEquivalentTo(email);
                     updatedCentre.CentrePostcode.Should().BeEquivalentTo(postcode);
+                    updatedCentre.ShowCentreOnMap.Should().BeFalse();
```
Comment (stellake) [inline DigitalLearningSolutions.Data.Tests/DataServices/CentresDataServiceTests.cs:215]: Nice! There are bunch of IDE warnings here, could we try and resolve them as part of this MR?

File: DigitalLearningSolutions.Data/DataServices/CentresDataService.cs [hunk 3]
```diff
@@ -108,6 +114,7 @@ FROM Centres
                             c.pwTelephone AS CentreTelephone,
                             c.pwEmail AS CentreEmail,
                             c.pwPostCode AS CentrePostcode,
+                            c.ShowOnMap AS ShowCentreOnMap,
```
Comment (stellake) [inline DigitalLearningSolutions.Data/DataServices/CentresDataService.cs:117]: I'd be happy for this to be simply "ShowOnMap" given that it lives on a data object.

**Review by stellake — APPROVED** [review]
Looks good, just two minor things (happy for it to be merged without rereview though):
1. The CentresDataServiceTests still has errors around vars (should be const string etc instead)
2. It looks like a few people have different HTML formatting defaults for leaving the space before "/>" - could we make it such that we leave a space before and add it to the editorconfig? 🙏

**Pull request merged**